File: Server/microservices/user-service/src/main/java/com/taskman/user_service/service/interfaces/TeamService.java
package com.taskman.user_service.service.interfaces;

import com.taskman.user_service.dto.TeamDTO;
import com.taskman.user_service.dto.TeamMembershipDTO;
import com.taskman.user_service.dto.UserDTO;
import com.taskman.user_service.dto.request.CreateTeamRequest;
import com.taskman.user_service.dto.request.UpdateTeamRequest;
import com.taskman.user_service.entity.enums.TeamRole;

import java.util.List;

public interface TeamService {
    // Create
    TeamDTO createTeam(CreateTeamRequest request);

    // Read
    TeamDTO getTeamById(Long id);
    List<TeamDTO> getAllTeams();
    List<TeamDTO> getTeamsByUserId(Long userId);
    List<TeamMembershipDTO> getTeamMembers(Long teamId);

    // Update
    TeamDTO updateTeam(Long id, UpdateTeamRequest request);

    // Delete
    void deleteTeam(Long id);

    // Membership management
    TeamMembershipDTO addMemberToTeam(Long teamId, Long userId, TeamRole role);
    void removeMemberFromTeam(Long teamId, Long userId);
    TeamMembershipDTO updateMemberRole(Long teamId, Long userId, TeamRole role);

    // Validation
    boolean isUserInTeam(Long teamId, Long userId);
    boolean existsByName(String name);

    // Team leaders
    List<UserDTO> getTeamLeaders(Long teamId);
}

File: taskman_front/src/services/TeamService.js
import api from "../utils/axios";

export const TeamService = {
  // Get all teams
  async getAllTeams() {
    const response = await api.get("/teams");
    return response.data;
  },

  // Get single team
  async getTeamById(teamId) {
    const response = await api.get(`/teams/${teamId}`);
    return response.data;
  },

  // Create new team
  async createTeam(teamData) {
    const response = await api.post("/teams", teamData);
    return response.data;
  },

  // Update team
  async updateTeam(teamId, teamData) {
    const response = await api.put(`/teams/${teamId}`, teamData);
    return response.data;
  },

  // Delete team
  async deleteTeam(teamId) {
    const response = await api.delete(`/teams/${teamId}`);
    return response.data;
  },

  async addMembers(teamId, data) {
    const response = await api.post(`/teams/${teamId}/members`, data);
    return response.data;
  },

  async removeMember(teamId, userId) {
    const response = await api.delete(`/teams/${teamId}/members/${userId}`);
    console.log(`api: /teams/${teamId}/members/${userId}`);
    return response.data;
  },
};

export default TeamService;
